testing/opencolorio: new aport

Klowner opened the pull request:
new package! 🌈

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,42 @@
+
+dev() {
+	default_dev
+}
```
`default_dev` is called implicitly when `dev` function is not declared, so you can just omit it.

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,42 @@
+build() {
+	cd "$builddir"
+	mkdir build && cd build
+	cmake -DCMAKE_INSTALL_PREFIX=/usr ..
```
Please add `|| return 1` at the end of this line, to ensure that an error will halt the build (abuilds don’t run with `set -e` yet).

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,42 @@
+	depends="python2"
+	mkdir -p "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/python2.7 "$subpkgdir"/usr/lib
+}
```
OpenColorIO supports both legacy Python and Python 3. Could you please also add a subpackage for Python 3?

Klowner replied:
It looks like I have to build it once for each python version (seems silly), I'll keep digging.

Klowner commented:
I think I've addressed all your feedback. It seems their build system isn't suited towards building python2 and python3 at the same time, so I'm just building both. If you'd prefer I take another route, let me know.

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,69 @@
+depends=""
+makedepends="cmake python2-dev python3-dev freeglut-dev glew-dev"
+install=""
+subpackages="$pkgname-python2:py2 $pkgname-python3:py3 $pkgname-dev $pkgname-utils"
```
Python subpackages should be named `py2-$pkgname` and `py3-$pkgname`.

jirutka commented:
Thanks for addressing the issues.

I’ve looked at opencolorio more closely. It bundles bunch of dependencies, namely lcms, tinyxml and yaml (and some others used only for document generation). Luckily these can be simply replaced with system-provided, you have to just pass these options to cmake:

```
-DUSE_EXTERNAL_YAML=TRUE
-DUSE_EXTERNAL_TINYXML=TRUE
-DUSE_EXTERNAL_LCMS=TRUE
```

… and delete the files, just to be sure: `rm -f ext/lcms* ext/tinyxml* ext/yaml*`.

We have a package for tinyxml and lcms, but not yet for yaml-cpp.

According to Fedora package [OpenColorIO](http://pkgs.fedoraproject.org/cgit/rpms/OpenColorIO.git/tree/OpenColorIO.spec?h=f25) it may be also needed to use `-DOCIO_USE_SSE=OFF` for non-x86_64 platforms (but I haven’t checked out OpenColorIO docs about this).

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,69 @@
+build() {
+	cd "$builddir"
+
+	for pyver in python3 python2; do
```
This variable should be declared as local (`local pyver`), to not pollute global namespace of the script.

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,69 @@
+}
+
+py2() {
+	PY2_VERSION=$(python -c 'import sys; print("%i.%i" % (sys.version_info.major, sys.version_info.minor))')
```
Maybe it’d be better (simpler) to use `python -c 'import site; print(site.getsitepackages()[0])'`, this directly prints location of the `site-packages` directory (e.g. `/usr/lib/python2.7/site-packages`).

jirutka replied:
Or you can entirely avoid it as in your second package using `mv "$pkgdir"/usr/lib/python$1?? "$subpkgdir"/usr/lib/` (or simply `$pkgdir"/usr/lib/python$1*`).

Klowner replied:
Oh yes, that's much nicer.

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,69 @@
+			-DOCIO_BUILD_TRUELIGHT=OFF \
+			-DOCIO_BUILD_NUKE=OFF \
+			|| return 1
+		make
```
Okay, I had to be more accurate previously. Every statement inside a function that may fail (and should not) should be followed by `|| return 1` (until we change abuild to run `set -e`). It’s not needed after the last statement of a function, because shell has implicit return. For example `cd ..`, `mkdir -p` etc. normally don’t fail, so not needed here, but `make` can fail and it should halt the built in that case.

Klowner commented:
Looks like they're using a `yaml-cpp > 0.3.0` which appears to be different from the `yaml-dev` in Alpine. I suppose I could make an `APKBUILD` for that too?

jirutka commented:
> I suppose I could make an APKBUILD for that too?

Yes, exactly. Since it’s a dependency of this package, add it to this pull request.

Klowner commented:
Think that's everything, do you see anything else I can improve?

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,77 @@
+depends=""
+makedepends="cmake python2-dev python3-dev freeglut-dev glew-dev lcms2-dev tinyxml-dev yaml-cpp-dev boost-dev"
+install=""
+subpackages="py2-$pkgname:py2 py3-$pkgname:py3 $pkgname-dev $pkgname-utils"
```
It’d be better to rename opencolorio-utils to opencolorio-tools. This name uses Fedora and Debian, so it’d be more clear for users coming from main-stream distros.

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,77 @@
+install=""
+subpackages="py2-$pkgname:py2 py3-$pkgname:py3 $pkgname-dev $pkgname-utils"
+source="$pkgname-$pkgver.tar.gz::http://github.com/imageworks/OpenColorIO/archive/v$pkgver.tar.gz
+	yaml-cpp-0.5-support.patch"
```
Please add also [this patch](https://github.com/imageworks/OpenColorIO/commit/1384029df00525c479e2c7e52fdfa294413fceb6) that fixes dereference of temporary when yaml-cpp 0.5.x is used.

jirutka commented:
I’ve merged testing/yaml-cpp in 47c4c32a7954941a3250746a1e4910ce55875910.

jirutka commented on testing/opencolorio/APKBUILD:
```diff
@@ -0,0 +1,77 @@
+# Contributor: Mark Riedesel <mark@klowner.com>
+# Maintainer:
```
Why you haven’t declared yourself as a maintainer?

Klowner replied:
I must have misunderstood the purpose of the maintainer field. The simple example ( https://wiki.alpinelinux.org/wiki/APKBUILD_examples:Simple ) also contains a blank maintainer field.

jirutka replied:
Aha. Have you used `newapkbuild` when creating abuild? This automatically fills Maintainer.

Maintainer is a person who’s responsible for maintaining the package. That person i.a. gets mails from http://pkgs.alpinelinux.org/ when someone flags the package as outdated or something. Maintainer can be anyone, not just Alpine core or trusted devs.

Klowner commented:
Good catch on finding that temp dereference patch. It's now included.
